### PR TITLE
CloudFormation, RiffRaff etc. to get the new account-frontend working in CODE

### PR DESCRIPTION
**Pull request by twrichards**
All the cloud-formation and riff-raff configuration to successfully deploy and run the application in CODE environment.

PLUS - a number of smaller fixes to ensure the application works...
* environment specific URLs
* correct resolution of static assets (which is ket to the server-side rendering)
* handling {} responses from members-data-api

### CODE REVIEW

File: cloud-formation/account-frontend-cf.yaml [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+      HealthCheckPath: /_healthcheck
+      HealthCheckPort: 9233
+      HealthCheckProtocol: HTTP
+      HealthCheckTimeoutSeconds: 5
+      HealthyThresholdCount: 2
+      UnhealthyThresholdCount: 2
+      VpcId: !Ref VpcId
+      TargetGroupAttributes:
+      - Key: deregistration_delay.timeout_seconds
+        Value: 45 # only connection drains for 45 seconds (rather than default of 300)
```
Comment (AWare) [inline cloud-formation/account-frontend-cf.yaml]: Let's just keep an eye on abandoned mda requests.

File: cloud-formation/account-frontend-cf.yaml [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+            # RUN
+            /usr/local/node-v9.11.1-linux-x64/bin/node /etc/gu/dist/server.js
+
+          - DomainEnvVariable: !FindInMap [ StageVariables, !Ref Stage, DomainEnvVariable ]
+
+  AppRole:
+    Type: AWS::IAM::Role
+    Properties:
+      Path: "/"
+      ManagedPolicyArns: [ "arn:aws:iam::aws:policy/service-role/AmazonEC2RoleforSSM" ]
```
Comment (jacobwinch) [inline cloud-formation/account-frontend-cf.yaml]: This role is overly permissive (I appreciate this is a mistake that was made elsewhere) - see https://github.com/guardian/support-frontend/pull/665

File: cloud-formation/account-frontend-cf.yaml [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+      AssumeRolePolicyDocument:
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Policies:
```
Comment (jacobwinch) [inline cloud-formation/account-frontend-cf.yaml]: We should not need the permissions to use fetch these keys from S3 (due to GDPR, we are changing the way that we control ssh access to instances).

File: cloud-formation/account-frontend-cf.yaml [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+  CreateProdResources: !Equals [!Ref "Stage", "PROD"]
+  CreateCodeResources: !Equals [!Ref "Stage", "CODE"]
+Mappings:
+  StageVariables:
+    CODE:
+      MaxInstances: 2
+      MinInstances: 1
+      InstanceType: t2.small
+      CertificateARN: arn:aws:acm:eu-west-1:865473395570:certificate/e4c03b58-bf76-44c2-ae6a-17f81c10eb6f
+      AwsKeyARN: arn:aws:kms:eu-west-1:865473395570:key/89f17663-c79e-4104-b718-ee0986402ad9
```
Comment (jacobwinch) [inline cloud-formation/account-frontend-cf.yaml]: Out of interest, what are we encrypting/decrypting?

**Pull request merged**